tests/foreman/cli/test_hostgroup.py

```diff
@@ -38,22 +38,40 @@
 from robottelo.cli.factory import make_subnet
 from robottelo.cli.hostgroup import HostGroup
 from robottelo.cli.proxy import Proxy
+from robottelo.cli.puppet import Puppet
 from robottelo.config import settings
 from robottelo.datafactory import invalid_id_list
 from robottelo.datafactory import invalid_values_list
 from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_hostgroups_list
 
-
 pytestmark = [
     pytest.mark.skipif(
         (not settings.robottelo.REPOS_HOSTING_URL), reason='Missing repos_hosting_url'
     )
 ]
-PUPPET_MODULES = [
-    {'author': 'robottelo', 'name': 'generic_1'},
-    {'author': 'robottelo', 'name': 'generic_2'},
-]
+
+pc_name = 'generic_1'
+
+
+@pytest.fixture(scope='module')
+def env(module_org, module_location, default_sat):
+    """Return the puppet environment."""
+
+    env_name = default_sat.create_custom_environment(repo=pc_name)
+    env = default_sat.api.Environment().search(query={'search': f'name={env_name}'})[0].read()
+    env = entities.Environment(
```

Comment:
This is re-creating the `env` entity that's searched for and read on the line above.

You should just be able to set fields on `env` before calling update and passing the field names.

Reply:
omkar I think, Mike ment usage like [this in api fixtures](https://github.com/SatelliteQE/robottelo/blob/a1c87aef7f8217e70152969f96fb1a7b18d822d4/pytest_fixtures/api_fixtures.py#L556-L558) 

or I think it should be possible to skip the search and make an update directly, and instead of `id` you can pass `name=env_name`  (didn't test it though)